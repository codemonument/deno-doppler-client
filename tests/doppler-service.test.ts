import { assertSnapshot, describe, it } from "std_testing";
import { DopplerService } from "@/mod.ts";
import { DOPPLER_TOKEN } from "@/env.ts";

const doppler = new DopplerService({ token: DOPPLER_TOKEN });

describe(`DopplerService`, () => {
  it(`reads configs (per environment)`, async (ctx) => {
    const configs = await doppler.getConfigs("deno-doppler-client");
    await assertSnapshot(ctx, configs);
  });

  it(`reads secrets from one config`, async (ctx) => {
    const secrets = await doppler.getSecrets("deno-doppler-client", "dev");
    await assertSnapshot(ctx, secrets);
  });

  it(`reads one secret from one config`, async (ctx) => {
    const secret = await doppler.retrieveSecret(
      "deno-doppler-client",
      "dev",
      "DEMO_SECRET",
    );
    await assertSnapshot(ctx, secret);
  });
});
