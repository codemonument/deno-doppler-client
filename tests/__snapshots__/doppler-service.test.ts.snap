export const snapshot = {};

snapshot[`DopplerService > reads configs (per environment) 1`] = `
{
  configs: [
    {
      created_at: "2022-10-14T21:33:16.593Z",
      environment: "dev",
      initial_fetch_at: "2022-10-14T23:24:47.458Z",
      last_fetch_at: "2022-10-14T23:55:48.958Z",
      locked: true,
      name: "dev",
      project: "deno-doppler-client",
      root: true,
      slug: "de5860bf-8d45-40d4-acb6-acb918c6a980",
    },
  ],
  page: 1,
  success: true,
}
`;

snapshot[`DopplerService > reads secrets from one config 1`] = `
{
  secrets: {
    DEMO_SECRET: {
      computed: "This is a demo secret! :)",
      raw: "This is a demo secret! :)",
    },
    DOPPLER_CONFIG: {
      computed: "dev",
      raw: "",
    },
    DOPPLER_ENVIRONMENT: {
      computed: "dev",
      raw: "",
    },
    DOPPLER_PROJECT: {
      computed: "deno-doppler-client",
      raw: "",
    },
  },
  success: true,
}
`;

snapshot[`DopplerService > reads one secret from one config 1`] = `
{
  name: "DEMO_SECRET",
  success: true,
  value: {
    computed: "This is a demo secret! :)",
    raw: "This is a demo secret! :)",
  },
}
`;
